platform/posix/clock_posix.c

```diff
@@ -63,8 +67,9 @@ void Clock_SleepMs( uint32_t sleepTimeMs )
 {
     /* Convert parameter to timespec. */
     struct timespec sleepTime = { 0 };
-    sleepTime.tv_sec = sleepTimeMs / MILLISECONDS_PER_SECOND;
-    sleepTime.tv_nsec = ( sleepTimeMs % MILLISECONDS_PER_SECOND ) * NANOSECONDS_PER_MILLISECOND;
+
+    sleepTime.tv_sec = ( ( time_t ) sleepTimeMs / ( time_t ) MILLISECONDS_PER_SECOND );
```

Comment:
Does this not work?
`sleepTime.tv_sec = ( time_t ) ( sleepTimeMs /  MILLISECONDS_PER_SECOND );`

Reply:
No. The issue was that the result of a composite expression cannot be casted to a different type. So that is not a solution.